eras/allegra/impl/src/Cardano/Ledger/Allegra/Rules/Utxo.hs

```diff
@@ -138,6 +138,14 @@ instance
 data AllegraUtxoEvent era
   = UpdateEvent (Event (EraRule "PPUP" era))
   | TotalDeposits (SafeHash (EraCrypto era) EraIndependentTxBody) Coin
+  | -- | The UTxOs consumed and created by a signal tx
+    TxUTxODiff
+      ( (UTxO era)
+      , -- \^ UTxOs consumed
+        (UTxO era)
+      )
+
+-- \^ UTxOs created
```

Comment:
Using a tuple is totally unnecessary. No only haddock can't be used to document the fields, but also tuples come with unnecessary overhead with respect to runtime and extra code that works operates on it
```suggestion
      (UTxO era)
      -- ^ UTxOs consumed
        (UTxO era)
      -- ^ UTxOs created
```

Alternatively, if you are worried (I would be) that the user of this event can mixup which ones are additions and which ones are deletions, we can create a separate type and use it here instead:
```haskell
data UTxODiff era = UTxODiff
  { utxoDiffConsumed :: !(UTxO era)
  , utxoDiffProduced :: !(UTXO era)
  }
```